frontend/src/controllers/PromotionFormController.es6

```diff
@@ -44,6 +45,16 @@ export default class {
         this.service.get(uuid)
             .then(this.transformDates.bind(self))
             .then(this.fillCampaignInfo.bind(self))
+            .then(p => {
+                if (this.$scope.createPromotionCopy) {
+                    return Object.assign({}, { ... p }, {
+                        uuid: this.uuid.v4(),
+                        name: `${p.name} [COPY]`,
```

Comment:
so when copy is hit, we go to the route "copy" with the uuid of the old promotion,. then we generate a new uuid and suitable name.  So that means we are in edit mode of a new channel.   And the `... p` bit is pulling all the promos from the old promotion.
Then somehow we are  somehow having `regenerateCodes` set which should hopefully overwrite them, but I can't see where it's set.
Also I feel like we should be doing all the making the copy correct in one place, rather than relying on it happening based on a flag, if possible.

Having said all this I've no idea about coding style for es6! I am trying to learn starting from this PR.. 😁

Reply:
@johnduffell pretty much all correct in your description of events.

Regarding your point "making the copy correct in one place", I see how this could be done, I could make the copy in the `PromotionFormController` (as I do now), including assigning regenerated codes to the `promotion.codes` then I can and pass down the `promotion.codes` to the `ChannelCodeController` as before - meaning no changes would be required to the `ChannelCodeController` (so I could remove the `regenerateCodes` flag i'm passing down). I might however need to expose the `generateSuggestedPromoCode` function so `PromotionFormController` could access it when generating codes.

Reply:
@johnduffell, I carried out this refactor as suggested